llm.js

```diff
@@ -7,6 +7,17 @@ class LLMProvider {
   async summarizeStream(prompt, onChunk) {
     throw new Error("summarizeStream method must be implemented.");
   }
+
+  async post(url, body) {
+    const headers = {'Content-Type': 'application/json'};
+    if (this.apiKey) headers.authorization = `Bearer ${this.apiKey}`;
+    return fetch(url, {method: 'POST', headers, body: JSON.stringify(body)});
+  }
+
+  async postJson(url, body) {
+    const r = await post(url, body);
+    return r.json();
```

Comment:
this should be 
```
return await r.json()
```
so that the caller does not have to await the returned promise, they just get the value